ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -305,6 +305,8 @@ private MessageWriteSetStateInStore checkWriteSetStateInStore(MessageWriteSet me
       throws StoreException {
     int existingIdenticalEntries = 0;
     for (MessageInfo info : messageSetToWrite.getMessageSetInfo()) {
+      //TODO IndexValue
+      //Why not looking for TTLUpdates here?  just seems to be looking for if it exists
       if (index.findKey(info.getStoreKey(), fileSpan,
           EnumSet.of(PersistentIndex.IndexEntryType.PUT, PersistentIndex.IndexEntryType.DELETE)) != null) {
```

Comment:
This is a very confusing usage. I can explain:

The findKey logic:
If EnumSet is DELETE and PUT,
For DELETE, findKey just searches for DELETE.
For PUT, findKey searches for TTL_UPDATE and PUT, then merge them.

If EnumSet is TTL_UPDATE, it just look for TTL_UPDATE and return(no backward searching for original PUT)

In summary, if the goal is to look for a final state(PUT, DELETE), use (PUT, DELETE) as findKey() input. 
if the goal is to look for a particular type: TTL_UPDATE, use TTL_UPDATE as findKey() input.

Looking forward to refactor.